app/lib/activitypub/activity/follow.rb

```diff
@@ -6,7 +6,13 @@ def perform
 
     return if target_account.nil? || !target_account.local? || delete_arrived_first?(@json['id'])
 
-    follow = @account.follow!(target_account)
-    NotifyService.new.call(target_account, follow)
+    follow_request = FollowRequest.create!(account: @account, target_account: target_account)
+
+    if target_account.locked?
```

Comment:
No but we do.

If we get a Follow and we are locked, we create a follow request and notify the user about the follow request. If we are not locked, we auto-accept the follow and notify the user about the follow.